vault/expiration.go

```diff
@@ -963,12 +963,18 @@ func (m *ExpirationManager) lazyRevokeInternal(ctx context.Context, leaseID stri
 }
 
 // should be run on a schedule. something like once a day, maybe once a week
-func (m *ExpirationManager) attemptIrrevocableLeasesRevoke() {
+func (m *ExpirationManager) attemptIrrevocableLeasesRevoke(removeIrrevocableLeaseAfter time.Duration) {
+	// determine if irrevocable leases should be evaluated for removal after revocation attempt
+	removeIrrevocableLeases := m.removeIrrevocableLeasesEnabled(removeIrrevocableLeaseAfter)
+
 	m.irrevocable.Range(func(k, v interface{}) bool {
 		leaseID := k.(string)
 		le := v.(*leaseEntry)
 
-		if le.ExpireTime.Add(time.Hour).Before(time.Now()) {
+		expiryBuffer := le.ExpireTime.Add(time.Hour)
```

Comment:
Just a question: can you explain why the code is adding an hour "buffer" to the lease's ExpireTime here? (If you know!) Only wondering since it will also affect our timing calculation for the new force-revoke below.